packages/trees/frontend/sirius-components-trees/src/treeitems/context-menu/DefaultMenuItem.tsx

```diff
@@ -10,166 +10,42 @@
  * Contributors:
  *     Obeo - initial API and implementation
  *******************************************************************************/
-import { gql, useLazyQuery, useMutation } from '@apollo/client';
-import { IconOverlay, useMultiToast } from '@eclipse-sirius/sirius-components-core';
-import { useImpactAnalysisDialog } from '@eclipse-sirius/sirius-components-impactanalysis';
+import { IconOverlay } from '@eclipse-sirius/sirius-components-core';
 import ListItemIcon from '@mui/material/ListItemIcon';
 import ListItemText from '@mui/material/ListItemText';
 import MenuItem from '@mui/material/MenuItem';
-import { useEffect } from 'react';
+import Tooltip from '@mui/material/Tooltip';
 import { DefaultMenuItemProps } from './DefaultMenuItem.types';
-import { useInvokeImpactAnalysis } from './impact-analysis/useTreeImpactAnalysis';
-import { GQLInvokeImpactAnalysisVariables } from './impact-analysis/useTreeImpactAnalysis.types';
-import {
-  GQLErrorPayload,
-  GQLFetchTreeItemContextEntryDataData,
-  GQLGetFetchTreeItemContextMenuEntryDataQueryVariables,
-  GQLInvokeSingleClickTreeItemContextMenuEntryData,
-  GQLInvokeSingleClickTreeItemContextMenuEntryInput,
-  GQLInvokeSingleClickTreeItemContextMenuEntryPayload,
-  GQLInvokeSingleClickTreeItemContextMenuEntryVariables,
-} from './TreeItemContextMenu.types';
 import { GQLTreeItemContextMenuEntry } from './useContextMenuEntries.types';
-
-const invokeSingleClickTreeItemContextMenuEntryMutation = gql`
-  mutation invokeSingleClickTreeItemContextMenuEntry($input: InvokeSingleClickTreeItemContextMenuEntryInput!) {
-    invokeSingleClickTreeItemContextMenuEntry(input: $input) {
-      __typename
-      ... on ErrorPayload {
-        message
-      }
-    }
-  }
-`;
-
-const getFetchTreeItemContextMenuEntryDataQuery = gql`
-  query getFetchTreeItemContextMenuEntryDataQuery(
-    $editingContextId: ID!
-    $representationId: ID!
-    $treeItemId: ID!
-    $menuEntryId: ID!
-  ) {
-    viewer {
-      editingContext(editingContextId: $editingContextId) {
-        representation(representationId: $representationId) {
-          description {
-            ... on TreeDescription {
-              fetchTreeItemContextMenuEntryData(treeItemId: $treeItemId, menuEntryId: $menuEntryId) {
-                urlToFetch
-                fetchKind
-              }
-            }
-          }
-        }
-      }
-    }
+import { useInvokeContextMenuEntry } from './useInvokeContextMenuEntry';
```

Comment:
The code to invoke a context menu entry is now in a hook to make it reusable here and in `useTreeKeyBinding`